### DISCUSSION TITLE
Configuring OpenID Connect Social Backend

### DISCUSSION THREAD

**Discussion by HadleySo** [inventree/InvenTree#5966] [Q&A]
My main question is how `social_providers` in `config.yml` gets sent to `SOCIALACCOUNT_PROVIDERS` in djago-allauth.

I'm trying to pass specific settings to my OpenID Connect social backend, following the documentation from [InvenTree](https://docs.inventree.org/en/latest/settings/SSO/#configuration-file) and [django-allauth](https://docs.allauth.org/en/latest/socialaccount/providers/openid_connect.html).

But I'm unable to get my OpenID Connect social application to appear in the InvenTree Admin panel or on the login screen. Microsoft Graph is shows up in the Admin panel and login screen. So far I've tried the following `config.yml` settings:

```
social_providers: 
    openid_connect: 
        APPS: [
            {
                "provider_id": "example-pro",
                "name": "OIDC Login",
                "client_id": "xxxxx",
                "secret": "xxxx",
                "settings": {
                    "server_url": "https://sso.example.com",
                    # Optional token endpoint authentication method.
                    # May be one of "client_secret_basic", "client_secret_post"
                    # If omitted, a method from the the server's
                    # token auth methods list is used
                    "token_auth_method": "client_secret_post",
                },
            },
        ]
    
```

and

```
SOCIALACCOUNT_PROVIDERS: {
    "openid_connect": {
        "APPS": [
            {
                "provider_id": "example-pro",
                "name": "OIDC Login",
                "client_id": "xxx",
                "secret": "xxx",
                "settings": {
                    "server_url": "https://sso.example.com",
                    # Optional token endpoint authentication method.
                    # May be one of "client_secret_basic", "client_secret_post"
                    # If omitted, a method from the the server's
                    # token auth methods list is used
                    "token_auth_method": "client_secret_basic",
                },
            },
        ]
    }
}
```

and
```

social_providers:
  openid_connect:
    id: oidc-pro
    name: OIDC Login

```

Any guidance on how to get OIDC setup would be greatly appreciated!

**Comment by SchrodingersGat** [comment]
Have you added OIDC under the `social_backends` section in your config file?

https://docs.inventree.org/en/stable/settings/SSO/?h=sso#configuration-file

**Reply by HadleySo** [reply]
Yup - checked the name against the Django allauth docs too.

Not getting any errors when restarting InvenTree and other backends show up in the admin panel, both with and without yaml config.

```
social_backends:
  - 'allauth.socialaccount.providers.microsoft'
  - 'allauth.socialaccount.providers.openid_connect'
```

**Comment by SchrodingersGat** [comment]
Related issue: https://github.com/inventree/InvenTree/issues/6037

**Comment by belotv** [comment]
Inventree uses the library in version 0.54 whereas openid_connect was introduced in 0.56.

**Reply by SchrodingersGat** [reply]
@belotv where are you getting this from? The changeling indicates it was introduced in `0.52.0`

https://docs.allauth.org/en/latest/release-notes/recent.html#id30

**Reply by belotv** [reply]
Let's say that it was not manageable through UI until 0.55 (this is clearly stated in the changelog) and was still buggy with some providers  (like Keycloak) until 0.56. That is why I would consider 0.56 to be the first version in which the implementation is complete enough.

Kr

**Reply by SchrodingersGat** [reply]
@belotv OK, I see. Will have to look into this - we did not port to 0.55 due to another bug it introduced with a different dependency, so it will have to be checked very carefully
